iris-mpc-cpu/src/hawkers/galois_store.rs

```diff
@@ -133,10 +139,34 @@ impl LocalNetAby3NgStoreProtocol {
     }
 }
 
+#[derive(Debug, Clone, Hash, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(bound = "")]
+pub struct DistanceShare<T: IntRing2k> {
+    code_dot: Share<T>,
+    mask_dot: Share<T>,
+    player:   Identity,
+}
+
+async fn eval_pairwise_distances(
+    mut pairs: Vec<(GaloisRingSharedIris, GaloisRingSharedIris)>,
+    player_session: &mut Session,
+) -> Vec<Share<u16>> {
+    pairs.iter_mut().for_each(|(_x, y)| {
+        y.code.preprocess_iris_code_query_share();
+        y.mask.preprocess_mask_code_query_share();
+    });
+    let ds_and_ts = galois_ring_pairwise_distance(player_session, &pairs)
+        .await
+        .unwrap();
+    galois_ring_to_rep3(player_session, ds_and_ts)
+        .await
+        .unwrap()
+}
+
 impl VectorStore for LocalNetAby3NgStoreProtocol {
     type QueryRef = PointId; // Vector ID, pending insertion.
     type VectorRef = PointId; // Vector ID, inserted.
-    type DistanceRef = (PointId, PointId); // Lazy distance representation.
+    type DistanceRef = Vec<DistanceShare<u16>>; // Distance represented as shares.
```

Comment:
This is the shared distance to all rotated copies of query?

Reply:
At the moment, it is only to one specific rotation (actually, to an original iris without rotations). We'll extend it later to pick the best distance among rotated copies.

Reply:
`Vec` here stands for shares of all parties (so it's length is 3).